demisto_sdk/commands/validate/validators/BA_validators/BA113_is_content_item_name_contain_trailing_spaces.py

```diff
@@ -0,0 +1,95 @@
+from demisto_sdk.commands.content_graph.objects.generic_type import GenericType
+from demisto_sdk.commands.content_graph.objects.incident_type import IncidentType
+from demisto_sdk.commands.content_graph.objects.indicator_field import IndicatorField
+from demisto_sdk.commands.content_graph.objects.integration import Integration
+from demisto_sdk.commands.content_graph.objects.layout import Layout
+from demisto_sdk.commands.content_graph.objects.layout_rule import LayoutRule
+from demisto_sdk.commands.content_graph.objects.mapper import Mapper
+from demisto_sdk.commands.content_graph.objects.modeling_rule import ModelingRule
+from demisto_sdk.commands.content_graph.objects.parsing_rule import ParsingRule
+from demisto_sdk.commands.content_graph.objects.playbook import Playbook
+from demisto_sdk.commands.content_graph.objects.report import Report
+from demisto_sdk.commands.content_graph.objects.script import Script
+from demisto_sdk.commands.content_graph.objects.test_playbook import TestPlaybook
+from demisto_sdk.commands.content_graph.objects.trigger import Trigger
+from demisto_sdk.commands.content_graph.objects.widget import Widget
+from demisto_sdk.commands.content_graph.objects.wizard import Wizard
+from demisto_sdk.commands.content_graph.objects.xsiam_dashboard import XSIAMDashboard
+from demisto_sdk.commands.content_graph.objects.xsiam_report import XSIAMReport
+from demisto_sdk.commands.validate.validators.base_validator import (
+    BaseValidator,
+    FixResult,
+    ValidationResult,
+)
+
+ContentTypes = Union[
+    Classifier,
+    CorrelationRule,
+    Dashboard,
+    GenericDefinition,
+    GenericField,
+    GenericModule,
+    GenericType,
+    IncidentType,
+    IndicatorField,
+    Integration,
+    Layout,
+    LayoutRule,
+    Mapper,
+    ModelingRule,
+    ParsingRule,
+    Playbook,
+    Report,
+    Script,
+    TestPlaybook,
+    Trigger,
+    Widget,
+    Wizard,
+    XSIAMDashboard,
+    XSIAMReport,
+]
+
+
+class IsContentItemNameContainTrailingSpacesValidator(BaseValidator[ContentTypes]):
+    error_code = "BA113"
+    description = "Validator to check if content item names contain trailing spaces."
+    error_message = (
+        "Content item name '{0}' should not have trailing spaces. Please remove."
+    )
+    fix_message = "Trailing spaces in the content item name '{0}' have been removed."
+    related_field = "name"
+    is_auto_fixable = True
+    expected_git_statuses = [GitStatuses.MODIFIED]
+    related_file_type = [RelatedFileType.YML]
+
+    def is_valid(self, content_items: Iterable[ContentTypes]) -> List[ValidationResult]:
+        return [
+            ValidationResult(
+                validator=self,
+                message=self.error_message.format(content_item.name),
+                content_object=content_item,
+            )
+            for content_item in content_items
+            if (content_item.name != content_item.name.strip())
```

Comment:
```suggestion
            if (content_item.name != content_item.name.rstrip())
```
more precise and matches the description